features/step_definitions/salt_steps.rb

```diff
@@ -376,3 +379,65 @@
 When(/^I select the state "(.*)"$/) do |state|
   find("input##{state}-cbox").click
 end
+
+# salt formulas
+When(/^I manually install the "([^"]*)" formula on the server$/) do |package|
+  $server.run("zypper --non-interactive install -y #{package}-formula", false)
+end
+
+When(/^I check the "([^"]*)" formula$/) do |formula|
+  # Complicated code because the checkbox is not a <input type=checkbox> but an <i>
+  xpath_query = "//a[@id = '#{formula}']/i[@class = 'fa fa-lg fa-square-o']"
+  if all(:xpath, xpath_query).any?
+    fail unless find(:xpath, xpath_query).click
+  else
+    xpath_query = "//a[@id = '#{formula}']/i[@class = 'fa fa-lg fa-check-square-o']"
+    assert all(:xpath, xpath_query).any?, "Checkbox could not be found"
+  end
+end
+
+Then(/^the "([^"]*)" formula should be checked$/) do |formula|
+  # Complicated code because the checkbox is not a <input type=checkbox> but an <i>
+  xpath_query = "//a[@id = '#{formula}']/i[@class = 'fa fa-lg fa-square-o']"
+  if all(:xpath, xpath_query).any?
+    fail "Checkbox is not checked"
+  end
+  xpath_query = "//a[@id = '#{formula}']/i[@class = 'fa fa-lg fa-check-square-o']"
+  assert all(:xpath, xpath_query).any?, "Checkbox could not be found"
+end
+
+When(/^I select "([^"]*)" in (.*) field$/) do |value, box|
+  boxid = case box
+    when "timezone name"
+      "timezone\$name"
+    when "language"
+      "keyboard_and_language\$language"
+    when "keyboard layout"
+      "keyboard_and_language\$keyboard_layout"
+  end
+  select(value, :from => boxid)
+end
+
+Then(/^the timezone on the minion should be "([^"]*)"$/) do |timezone|
```

Comment:
we should use here sle-minion in sake of readibility.
Because $minion is sle-minion, otherwise for super-generic you should use the " " for taking input minion. But we don't need super-generic there